index.bs

```diff
@@ -6128,7 +6128,7 @@ Given the following build graph:
     // Create MLOperandDescriptor object.
     const desc = {
       dataType: 'float32',
-      dimensions: TENSOR_DIMS
+      shape: TENSOR_DIMS
```

Comment:
`TENSOR_DIMS` -> `TENSOR_SHAPE`

(or maybe better, `TENSOR_DIMS` -> `TENSOR_SIZES`, which has the nice benefit that `TENSOR_SIZES` applies to the multidimensional case, and the existing `TENSOR_SIZE` definition below it applies to the 1D case of linear elements, both of which equal the same product)

Reply:
Updated to use `TENSOR_SHAPE` 👍 

> 🤔 Have you considered `sizes`?

Hmm I don't like `sizes` for the same reason I don't like `dimensions`:

> since "dimensions" - while referring to _one set of dimensions_ (i.e. one shape) - is plural, referring to a collection of dimensions is awkward and results in use of variable names like `dimensionsArray`, rather than just `shapes`

And since the "size" of an object generally refers to its _total size_, `TENSOR_SIZE` would become ambiguous. Is it referring to a single dimension (the size of a size), or the entire tensor (the size of the sizes)? 🙃

Reply:
> the size of the sizes?

ML libraries differ on many aspects, but I've never seen them diverge or be ambiguous on this. A tensor's size always means the total number of elements (just like you're used to with `std::vector`), never the extent of a single dimension. A tensor's *sizes* (in those libraries that use sizes) refers to the *list* of extents, with the total size being the product of sizes.

Reply:
\<ignorable-p-s\>Apparently the word `figure` means something's shape *and* size, but `tensor.figure` sounds kinda weird, and another word I just recalled (that we actually use inside DML a few places, but forgot) is the `extent`, "the degree to which something has spread; the size or scale of something". The `tensor.extent` doesn't suffer your plurality concern, and unlike shape, it actually means the sizes too, not just the ratio of the sizes. Maybe I'll use extent for my own ML toy library 😉. (I am reminded of The Giver's quote "Precision of language").\<ignorable-p-s\>